core/src/jsMain/kotlin/Transaction.kt

```diff
@@ -23,6 +24,26 @@ public open class Transaction internal constructor(
         }
     }
 
+    internal suspend fun awaitFailure() {
+        transaction.onNextEvent("complete", "abort", "error") { event ->
+            when (event.type) {
+                "abort" -> Unit
+                "error" -> Unit
+                else -> Unit
+            }
+        }
+    }
+    internal fun abort() {
+        transaction.abort()
+    }
+
+    internal fun commit() {
+        // Check if function exists before calling it.
```

Comment:
*note*:  commit was added later to browser, so it might not be always available 

https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction#browser_compatibility